testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -241,6 +270,90 @@ impl BlockMinerThread {
+            .get_signers(&signers_contract_id)
+            .expect("FATAL: could not get signers from stacker DB")
+            .iter()
+            .enumerate()
+            .map(|(id, _)| id as u32 * SIGNER_SLOTS_PER_USER + BLOCK_SLOT_ID)
+            .collect::<Vec<u32>>();
+        // If more than a threshold percentage of the signers reject the block, we should not wait any further
+        let rejection_threshold =
+            slot_ids.len() / 100 * self.config.miner.signer_rejection_threshold;
+        let mut num_rejections = 0;
+        let now = Instant::now();
+        while now.elapsed() < self.config.miner.wait_on_signers {
+            // Get the block responses from the signers for the block we just proposed
+            let signer_messages: Vec<SignerMessage> = stackerdbs
+                .get_latest_chunks(&signers_contract_id, &slot_ids)
+                .expect("FATAL: could not get latest chunks from stacker DB")
+                .into_iter()
+                .filter_map(|chunk| {
+                    chunk.and_then(|chunk| bincode::deserialize::<SignerMessage>(&chunk).ok())
+                })
+                .collect();
+            for signer_message in signer_messages {
+                match signer_message {
+                    SignerMessage::BlockResponse(BlockResponse::Accepted(block)) => {
+                        if block.header.signer_signature_hash().ok()? == *signer_signature_hash
+                            && block
+                                .header
+                                .signer_signature
+                                .verify(aggregate_public_key, &signer_signature_hash.0)
+                        {
+                            // We verified that the returned block contains a valid signature across the signer signature hash of our original block request
+                            // Let's just immediately attempt to append the block without waiting for more signer messages
+                            return Some(block);
+                        }
+                    }
+                    SignerMessage::BlockResponse(BlockResponse::Rejected(block_rejection)) => {
+                        // First check that this block rejection is for the block we proposed
+                        if block_rejection.block.header.signer_signature_hash().ok()?
+                            == *signer_signature_hash
+                        {
+                            let signature = block_rejection.block.header.signer_signature;
+                            let mut message = signer_signature_hash.0.to_vec();
+                            message.push(b'n');
+                            if signature.verify(aggregate_public_key, &message) {
+                                // We received a verified rejection. We will NEVER get a signed block from the signers for this particular block
+                                return None;
+                            } else if signature
```

Comment:
Why even count unverified rejections? If signers can't get an aggregate signature for either accept or reject, then no blocks could be produced anyway.

Reply:
We need to verify the rejection as a malicious signer could spam the miner with false rejections forcing them to remine a new block when they should not have to.